en/mavgen_c/message_signing_c.md

```diff
@@ -0,0 +1,125 @@
+# C Message Signing (mavgen)
+
+One of the key features of [MAVLink 2](../guide/mavlink_2.md) is support for [Message Signing](../guide/message_signing.md) (authentication).
+
+The C libraries generated by *mavgen* provide almost everything needed to support signing in your MAVLink system.
+You will need to add some code to:
+
+* Handle the [SETUP_SIGNING](../messages/common.md#SETUP_SIGNING) message.
+* Setup and teardown signing on a link.
+* Save and load the secret key and timestamp in persistent storage
+* Implement a callback to define which (if any) unsigned messages will be accepted.
+
+
+## Handling SETUP_SIGNING
+
+The [SETUP_SIGNING](../messages/common.md#SETUP_SIGNING) message is the mechanism for a GCS to setup a signing key on a *MAVLink 2* device. 
+It takes a 32 byte secret key and an initial timestamp. 
+The method of generating the 32 byte secret key is up to the GCS implementation, although it is suggested that all GCS implementations should support the use of a sha256 hash of a user provided passphrase.
+
+## Handling Timestamps
+
+The timestamp is a 64 bit number, and is in units of 10 microseconds since 1st January 2015. For systems where the time since 1/1/1970 is available (the unix epoch) you can use an offset in seconds of 1420070400.
```

Comment:
I think we need to properly define the mavlink epoch somewhere. Currently, we randomly use several other random times. 

For example : 
1. https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_messages.cpp#L1519-L1524
2. https://github.com/mavlink/mavros/blob/master/mavros/src/plugins/sys_time.cpp#L287
(There are more)

Reply:
AFAIK MAVLink doesn't have an epoch time. The first example is trying to determine whether a timestamp is Unix epoch or device local time. The second appears to be checking if a system time is likely to be valid. 

This timestamp referred to here wouldn't be MAVLink epoch time either - though I guess it could be. I presume it was chosen to be the time that signing was introduced.